### PR TITLE
[Shader Graph][HDRP] Enable Normal and Tangent Modification in Vertex 

### PR DESCRIPTION
**Pull request by alindmanUnity**
### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?
This pr enables the modification of normals and tangent in the vertex stage of a shader in shader graph. 

modNormal_corrected
(New behavior on the left, old behavior on the right)

---
### Release Notes
Please add any useful notes about the feature/fix that might be helpful for others.
Adds `Vertex Normal` and `Vertex Tangent` slots to the following HDRP-only master nodes:
- Lit
- Stack Lit
- HD Unlit 
- Eye
- Fabric
- Hair
- Decal

---
### Testing status
**Katana Tests**: First off we need to make sure the Katana SRP tests are green?

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally
- [x] Checked new UI names with UX convention
- [x] Tested UI multi-edition + Undo/Redo
- [x] C# and shader warnings (supress shader cache to see them)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

Launch Katana (Link on master here - update for your branch):
https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=master&automation-tools_branch=master&unity_branch=trunk&sort=1-asc

Alternative, launch Yamato (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/sg%252Fhdrp-vert-mod
Any test projects to go with this to help reviewers?

---
### Overall Product Risks
**Technical Risk**: None, Low, Medium, High? Low - Does not change any existing behavior, adds new functionality and interpolations 

**Halo Effect**: None, Low, Medium, High? Low - shouldn't affect existing graphs in any way 

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Editor/Material/Fabric/ShaderGraph/FabricSubShader.cs [hunk 1]
```diff
@@ -346,7 +356,9 @@ class FabricSubShader : IFabricSubShader
             },
             VertexShaderSlots = new List<int>()
             {
-                HDLitMasterNode.PositionSlotId
+                HDLitMasterNode.PositionSlotId,
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Editor/Material/Fabric/ShaderGraph/FabricSubShader.cs]: Looks like there is a mistake in original code, replce HDLitMasterNode by FabricMasterNode please (sorry I know it is not from your PR, but let's correct it at the same time)

File: com.unity.render-pipelines.high-definition/Editor/Material/Fabric/ShaderGraph/FabricSubShader.cs [hunk 2]
```diff
@@ -385,7 +397,9 @@ class FabricSubShader : IFabricSubShader
...
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Editor/Material/Fabric/ShaderGraph/FabricSubShader.cs]: Looks like there is a mistake in original code, replce HDLitMasterNode by FabricMasterNode please (sorry I know it is not from your PR, but let's correct it at the same time)

File: com.unity.render-pipelines.high-definition/Editor/Material/Fabric/ShaderGraph/FabricSubShader.cs [hunk 3]
```diff
@@ -431,7 +445,9 @@ class FabricSubShader : IFabricSubShader
...
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Editor/Material/Fabric/ShaderGraph/FabricSubShader.cs]: Looks like there is a mistake in original code, replce HDLitMasterNode by FabricMasterNode please (sorry I know it is not from your PR, but let's correct it at the same time)

File: com.unity.render-pipelines.high-definition/Editor/Material/Fabric/ShaderGraph/FabricSubShader.cs [hunk 4]
```diff
@@ -477,7 +493,9 @@ class FabricSubShader : IFabricSubShader
...
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Editor/Material/Fabric/ShaderGraph/FabricSubShader.cs]: Looks like there is a mistake in original code, replce HDLitMasterNode by FabricMasterNode please (sorry I know it is not from your PR, but let's correct it at the same time)

File: com.unity.render-pipelines.high-definition/Editor/Material/Unlit/ShaderGraph/HDUnlitSubShader.cs [hunk 5]
```diff
@@ -71,7 +71,9 @@ class HDUnlitSubShader : IHDUnlitSubShader
             },
             VertexShaderSlots = new List<int>()
             {
-                HDUnlitMasterNode.PositionSlotId
+                HDUnlitMasterNode.PositionSlotId,
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Editor/Material/Unlit/ShaderGraph/HDUnlitSubShader.cs]: unlit shader don't receive lighting, so it shouldn't have a need for modifying the tangent. (as there is no normal mapping). However modify the vertex normal can be used for tesselation / displacement. 
I guess we shouldn't expose tangent then. Maybe this need to be discuss and there is usage I don't think of. We can do this in another PR.

Comment (alindmanUnity) [reply com.unity.render-pipelines.high-definition/Editor/Material/Unlit/ShaderGraph/HDUnlitSubShader.cs]: It's a good interpolator to have for custom lighting scenarios, and it keeps the implementation consistent across the master nodes.

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Editor/Material/Unlit/ShaderGraph/HDUnlitSubShader.cs]: That's my point, Unlit don't have lighting :), so no custom lighting.

**Comment by alindmanUnity** [comment]
tests on yamato are green aside from known failure in master on HDRP standalone

**Comment by sebastienlagarde** [comment]
Let's move forward :)
